### PR TITLE
feat(mosaicod)!: Improve do_action tests scenarios

### LINKED ISSUE #427: Improve do_action tests scenarios in mosaicod
Not all existing tests for backend do_action cover all possible scenarios.

The following tests are missing specific scenarios that should be added:

- [x] `topic_delete` / `sequence_delete`: missing scenario of deleting `sequnce/topic` that doesn't exist
- [x] `topic_create`: missing scenario of malformed payload
- [x] `topic_create` / `sequence_create`: missing scenario of creating `sequnce/topic` that already exists
- [ ] topic is deleted before its session is finalized
- [ ] topic is deleted before uploading any data (no do_put)
- [ ] `do_get`

### PR DESCRIPTION
**Pull request by LuigiTer**
* Added more tests to improve edge cases and other missing cases.
* Fixed do_get on an empty topic returns Internal instead of FailedPrecondition
* Fixed Concurrent session_finalize calls produce inconsistent state

Closes https://github.com/mosaico-labs/mosaico/issues/427 https://github.com/mosaico-labs/mosaico/issues/469 https://github.com/mosaico-labs/mosaico/issues/468

### CODE REVIEW

File: mosaicod/crates/mosaicod-db/src/sql/pg_queries/session_record.rs [hunk 1]
```diff
@@ -79,16 +79,16 @@ pub async fn session_find_by_locator(
...
 }
 
-/// Returns true if the session is locked.
-pub async fn session_locked(exe: &mut impl AsExec, session_id: i32) -> Result<bool, Error> {
+/// Returns true if the session has already been finalized.
+pub async fn session_finalized(exe: &mut impl AsExec, session_id: i32) -> Result<bool, Error> {
     trace!("session (id=`{}`) locked? ", session_id);
-    let locked = sqlx::query_scalar!(
-        r#"SELECT (completion_unix_tstamp IS NOT NULL) AS "locked!" FROM session_t WHERE session_id=$1"#,
+    let finelized = sqlx::query_scalar!(
```
Comment (DevMattG) [inline mosaicod/crates/mosaicod-db/src/sql/pg_queries/session_record.rs]:
```suggestion
    let finalized = sqlx::query_scalar!(
```

File: mosaicod/crates/mosaicod-db/src/sql/pg_queries/session_record.rs [hunk 2]
```diff
@@ -79,16 +79,16 @@ pub async fn session_find_by_locator(
...
+        r#"SELECT (completion_unix_tstamp IS NOT NULL) AS "finalized!" FROM session_t WHERE session_id=$1"#,
         session_id
     )
         .fetch_one(exe.as_exec())
         .await?;
-    Ok(locked)
+    Ok(finelized)
```
Comment (DevMattG) [inline mosaicod/crates/mosaicod-db/src/sql/pg_queries/session_record.rs]:
```suggestion
    Ok(finalized)
```

**Review by DevMattG — CHANGES_REQUESTED** [review]

**Review by DevMattG — APPROVED** [review]
